core/src/main/scala/org/apache/spark/rdd/RDDCheckpointData.scala

```diff
@@ -66,6 +88,27 @@ private[spark] class RDDCheckpointData[T: ClassTag](@transient rdd: RDD[T])
     cpFile
   }
 
+  // Get the iterator used to write checkpoint data to HDFS
+  def getCheckpointIterator(
+      rddIterator: Iterator[T],
+      context: TaskContext,
+      partitionId: Int): Iterator[T] = {
+    RDDCheckpointData.synchronized {
+      if (cpState == Initialized) {
+        // Create the output path for the checkpoint
+        val path = new Path(checkpointDir.get, "rdd-" + rddId)
+        CheckpointingIterator[T](
+          rddIterator,
+          path.toString,
+          broadcastedConf,
+          partitionId,
+          context)
+      } else {
+        rddIterator
```

Comment:
This seems like an error condition. If we are not in initialized, then we shouldn't be doing this. I would throw an exception here instead.